### PR TITLE
Make defaultBaseImages configurable. Fixes #430

### LINKED ISSUE #430: Make `defaultBaseImage` configurable for prometheus and alertmanager
It would be useful to make these images configurable at a cluster level.

We provide a quay mirror inside all of our Kubernetes clusters to save bandwidth, and make them resilient to loss of internet. Currently we can override the `baseImage` in every `Prometheus` / `AlertManager` resource, but we have to tell all the people using the cluster to do so.

If we had a command line argument (like `--prometheus-config-reloader`), we could set this on the cluster level and life would be easier for our users.

### PR DESCRIPTION
**Pull request by mikebryant**

### CODE REVIEW

**Comment by brancz** [comment]
@coreosbot test this please

**Comment by brancz** [comment]
hmm .. I can't see why the tests failed there, I checked it out locally and it unit tests passed. I'll re-run. @coreosbot test this please.

File: pkg/alertmanager/statefulset.go [hunk 1]
```diff
@@ -48,7 +47,7 @@ func makeStatefulSet(am *v1alpha1.Alertmanager, old *v1beta1.StatefulSet, config
 	// Potentially an update handler on first insertion.
 
 	if am.Spec.BaseImage == "" {
-		am.Spec.BaseImage = defaultBaseImage
+		am.Spec.BaseImage = config.AlertmanagerDefaultBaseImage
```
Comment (brancz) [inline pkg/alertmanager/statefulset.go]: When instantiating the Alertmanager operator/controller, the Prometheus config is passed, but an Alertmanager config is instantiated from that config. The Alertmanager config needs to be adapted.

**Comment by mikebryant** [comment]
I ran the tests locally this time, and they passed.

*hangs head in shame*

Should have done that the first time, sorry.

**Comment by brancz** [comment]
No worries, that happens. @coreosbot test this please.

**Comment by brancz** [comment]
lgtm, thanks a lot @mikebryant !

**Pull request merged**